cmd/prchecker/main.go

```diff
@@ -41,79 +91,169 @@ func main() {
+// expression. If the expression matches then spanning modules are allowed.
+func ModuleSpanAllowed(repository GitHubService, pullId int) (bool, error) {
+
+	// Note: There are multiple ways to pull a github commit object
+	// we want a RepositoryCommit.
+	pullRequest, _, err := repository.GetPullRequest(pullId)
+
+	if err != nil {
+		return false, err
+	}
+
+	return StringAllowsModuleSpan(*pullRequest.Body)
+}
+
+// GetCommitChanges looks up a github commit by SHA and returns a Changeset
+// containing the modified files in the specified commit.
+func GetCommitChanges(repository GitHubService, commitSha string) (*Changeset, error) {
+
+	// Note: There are multiple ways to pull a github commit object
+	// we want a RepositoryCommit.
+	commit, _, err := repository.GetCommit(commitSha)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return ConstructChangeset(commit), nil
+}
+
+// GetPullRequestCommits constructs a list of all commits and the associated
+// files changes in the given pull request
+func GetPullRequestCommits(repository GitHubService, pullrequest int) ([]*Changeset, error) {
+
 	// foundFiles across all pages from github api
-	var foundFiles []*github.CommitFile
-	// GitHub returns the first 30 files by default, increase this value
-	// Note: Page 1 is the first page of results. Page 0 is an end of list mark.
-	// Github only returns (max) 100 results per page and PR's may exceed this
-	// so loop until all pages have been enumerated.
-	options := &github.ListOptions{PerPage: 100, Page: 1}
+	var collectedCommits []*github.RepositoryCommit
+	// Github only returns a limited set of commits per request and PR's may
+	// exceed this so loop until all pages have been enumerated.
+	options := &github.ListOptions{Page: 1}
 	for options.Page != 0 {
-		files, response, err := client.PullRequests.ListFiles(context.Background(), *owner, *repo, *pullrequest, options)
+		commits, response, err := repository.ListCommits(pullrequest, options)
 
 		// If an error has occurred while querying api exit early, report error
 		if err != nil {
 			return nil, err
 		}
-		foundFiles = append(foundFiles, files...)
+		collectedCommits = append(collectedCommits, commits...)
 		// setup next page to continue loop
-		options = &github.ListOptions{PerPage: 100, Page: response.NextPage}
+		options = &github.ListOptions{Page: response.NextPage}
 	}
-	return foundFiles, nil
-}
 
-// CountModifiedRestrictedDirectories Accepts a map of paths and the number of
-// occurances and returns the count of the paths which had a non-zero value.
-func CountModifiedRestrictedDirectories(contributedRestrictedPaths map[string]int) int {
-	modifiedRestrictedDirectories := 0
-	for _, occurance := range contributedRestrictedPaths {
-		if occurance != 0 {
-			modifiedRestrictedDirectories++
+	var changesetResults []*Changeset
+	for _, commit := range collectedCommits {
+		// The repository commits from list commits are not hydrated
```

Comment:
Perhaps not the right word, but this is meant to describe the API returning a partial object initially. Because many of the fields are null values they need to be populated by requesting the values explicitly with another API call.